pkg/resources/row_access_policy_acceptance_test.go

```diff
@@ -32,54 +42,393 @@ func TestAcc_RowAccessPolicy(t *testing.T) {
+					resourceshowoutputassert.RowAccessPolicyShowOutput(t, resourceName).
+						HasCreatedOnNotEmpty().
+						HasDatabaseName(id.DatabaseName()).
+						HasKind(string(sdk.PolicyKindRowAccessPolicy)).
+						HasName(id.Name()).
+						HasOptions("").
+						HasOwner(snowflakeroles.Accountadmin.Name()).
+						HasOwnerRoleType("ROLE").
+						HasSchemaName(id.SchemaName()).
+						HasComment("Terraform acceptance test"),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.body", "case when current_role() in ('ANALYST') then true else false end")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.name", id.Name())),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.return_type", "BOOLEAN")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.signature", "(N VARCHAR, V VARCHAR)")),
 				),
 			},
 			// change comment and expression
 			{
 				ConfigDirectory: config.TestStepDirectory(),
 				ConfigVariables: m(),
-				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "name", id.Name()),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "fully_qualified_name", id.FullyQualifiedName()),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "database", acc.TestDatabaseName),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "schema", acc.TestSchemaName),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "comment", "Terraform acceptance test - changed comment"),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "row_access_expression", "case when current_role() in ('ANALYST') then false else true end"),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "signature.N", "VARCHAR"),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "signature.V", "VARCHAR"),
+				Check: assert.AssertThat(t, resourceassert.RowAccessPolicyResource(t, resourceName).
+					HasNameString(id.Name()).
+					HasDatabaseString(id.DatabaseName()).
+					HasSchemaString(id.SchemaName()).
+					HasFullyQualifiedNameString(id.FullyQualifiedName()).
+					HasCommentString("Terraform acceptance test - changed comment").
+					HasBodyString("case when current_role() in ('ANALYST') then false else true end"),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.#", "2")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.0.name", "N")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.0.type", "VARCHAR")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.1.name", "V")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.1.type", "VARCHAR")),
 				),
 			},
 			// change signature
```

Comment:
please change the singature externally as a separate step (in a few ways); I think that it would be best to have a separate test case for just signature modifications: in config and external

Reply:
Done.